src/kafka/ztf.rs

```diff
@@ -120,134 +133,219 @@ impl AlertConsumer for ZtfAlertConsumer {
-        .arg(&url)
-        .arg("-P")
-        .arg(&data_folder)
-        .arg("-O")
-        .arg(format!("{}/ztf_public_{}.tar.gz", data_folder, date))
-        .output()?;
-    if !output.status.success() {
-        let error_msg = String::from_utf8_lossy(&output.stderr);
-        if error_msg.contains("404 Not Found") {
-            return Err("No alerts found for this date".into());
+impl ZtfAlertProducer {
+    pub fn download_alerts_from_archive(&self) -> Result<i64, Box<dyn std::error::Error>> {
+        if self.date.len() != 8 {
+            return Err("Invalid date format".into());
         }
-        return Err(format!("Failed to download alerts: {}", error_msg).into());
-    } else {
-        info!("Downloaded alerts to {}", data_folder);
-    }
-
-    let output = std::process::Command::new("tar")
-        .arg("-xzf")
-        .arg(format!("{}/ztf_public_{}.tar.gz", data_folder, date))
-        .arg("-C")
-        .arg(&data_folder)
-        .output()?;
-    if !output.status.success() {
-        return Err("Failed to extract alerts".into());
-    } else {
-        info!("Extracted alerts to {}", data_folder);
-    }
-
-    std::fs::remove_file(format!("{}/ztf_public_{}.tar.gz", data_folder, date))?;
 
-    let count = std::fs::read_dir(&data_folder)?.count();
+        info!(
+            "Downloading alerts for date {} (programid: {})",
+            self.date, self.program_id
+        );
 
-    Ok(count as i64)
-}
+        let (file_name, data_folder, base_url) = match self.program_id {
+            1 => (
+                format!("ztf_public_{}.tar.gz", self.date),
+                format!("data/alerts/ztf/public/{}", self.date),
+                "https://ztf.uw.edu/alerts/public/".to_string(),
+            ),
+            2 => (
+                format!("ztf_partnership_{}.tar.gz", self.date),
+                format!("data/alerts/ztf/partnership/{}", self.date),
+                "https://ztf.uw.edu/alerts/partnership/".to_string(),
+            ),
+            _ => return Err("Invalid program ID".into()),
+        };
+
+        std::fs::create_dir_all(&data_folder)?;
+
+        if std::fs::read_dir(&data_folder)?.count() > 0 {
+            info!("Alerts already downloaded to {}", data_folder);
+            let count = std::fs::read_dir(&data_folder).unwrap().count();
+            return Ok(count as i64);
+        }
 
-pub async fn produce_from_archive(
-    date: &str,
-    limit: i64,
-    topic: Option<String>,
-) -> Result<i64, Box<dyn std::error::Error>> {
-    match download_alerts_from_archive(&date) {
-        Ok(count) => count,
-        Err(e) => {
-            error!("Error downloading alerts: {}", e);
-            return Err(e);
+        let mut command = std::process::Command::new("wget");
+        command
+            .arg(&format!("{}{}", base_url, file_name))
+            .arg("-P")
+            .arg(&data_folder)
+            .arg("-O")
+            .arg(format!("{}/{}.temp", data_folder, file_name));
```

Comment:
This temp file path formatting is repeated 3 times. Recommend setting as a variable. Also might want to use [tempfile](https://docs.rs/tempfile/latest/tempfile/).

Reply:
Let me make it a variable

Reply:
+1 for tempfile

Reply:
One thing that bothers me with tempfile is that it downloads - by default - to the system's temp file directory. I'm thinking of a docker deploy here, but I may not want to download data in a random dir in my container but instead would like it to be where my data folder has been mounted. So I'm fine using tempfile as long as I can specify what dir to put that tempfile in. Let me look at the docs.

Reply:
also right now I create a temp file then rename it to another name (i.e. move it), is that what tempfile really is for?

Reply:
though I could perhaps skip that step...